lib/measured/conversion_builder.rb

```diff
@@ -0,0 +1,35 @@
+class Measured::ConversionBuilder
+  def initialize
+    @base_unit = nil
+    @units = []
+    @case_sensitive = false
+  end
+
+  def base(unit_name, aliases: [])
+    raise Measured::UnitError, "Setting #{unit_name} as base unit, but already defined as #{@base_unit}." if @base_unit
+    @base_unit = Measured::Unit.new(unit_name, aliases: aliases)
+  end
+
+  def unit(unit_name, aliases: [], value:)
+    @units << Measured::Unit.new(unit_name, aliases: aliases, value: value)
+  end
+
+  def case_sensitive(is_case_sensitive)
+    @case_sensitive = is_case_sensitive
+  end
+
+  def conversion
+    raise Measured::UnitError, "A base unit has not yet been set." unless @base_unit
+    check_for_duplicate_unit_names
```

Comment:
The former is satisfied, but the latter would not be satisfied if someone decided to use `ConversionBuilder` without calling conversion. I'd love to make `ConversionBuilder` a private detail of this gem, but we're in Ruby and can't do that. I'll switch the check back to be performed each time a unit is added.